lib/ansible/modules/cloud/google/gcp_compute_region.py

```diff
@@ -0,0 +1,235 @@
+#
+#     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+#
+# ----------------------------------------------------------------------------
+#
+#     This file is automatically generated by Magic Modules and manual
+#     changes will be clobbered when the file is regenerated.
+#
+#     Please read more about how to change this file at
+#     https://www.github.com/GoogleCloudPlatform/magic-modules
+#
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+################################################################################
+# Documentation
+################################################################################
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ["preview"],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: gcp_compute_region
+description:
+    - Represents a Region resource. A region is a specific geographical
+      location where you can run your resources. Each region has one or more
+      zones
+short_description: Creates a GCP Region
+version_added: 2.6
+author: Google Inc. (@googlecloudplatform)
+requirements:
+    - python >= 2.6
+    - requests >= 2.18.4
+    - google-auth >= 1.3.0
+options:
+    state:
+        description:
+            - Whether the given object should exist in GCP
+        required: true
+        choices: ['present', 'absent']
+        default: 'present'
+    name:
+        description:
+            - Name of the resource.
+        required: false
+extends_documentation_fragment: gcp
+'''
+
+EXAMPLES = '''
+- name: verify a region
+  gcp_compute_region:
+      name: 'us-west1'
+      project: testProject
+      auth_kind: service_account
+      service_account_file: /tmp/auth.pem
+      scopes:
+        - https://www.googleapis.com/auth/compute
+      state: present
+'''
+
+RETURN = '''
+    creation_timestamp:
+        description:
+            - Creation timestamp in RFC3339 text format.
+        returned: success
+        type: str
+    deprecated_deleted:
+        description:
+            - An optional RFC3339 timestamp on or after which the deprecation
+              state
+              of this resource will be changed to DELETED.
+        returned: success
+        type: str
+    deprecated_deprecated:
+        description:
+            - An optional RFC3339 timestamp on or after which the deprecation
+              state
+              of this resource will be changed to DEPRECATED.
+        returned: success
+        type: str
+    deprecated_obsolete:
+        description:
+            - An optional RFC3339 timestamp on or after which the deprecation
+              state
+              of this resource will be changed to OBSOLETE.
+        returned: success
+        type: str
+    deprecated_replacement:
```

Comment:
I think these returns (`deprecated_*`) should be under an object along the lines of:

```
deprecated:
  state: [one of OBSOLETE, DELETED, DEPRECATED]
  obsolete_at: [ISO stamp]
  deprecated_at: [ISO stamp]
  deleted_at: [ISO stamp]
```

That way users can do checks like `{{ if mything.deprecated }}` and such, instead of the snake-cased returns, and you can return mything.deprecated==None if there's no deprecation yet.